pajbot/modules/wide_emote_limit.py

```diff
@@ -0,0 +1,125 @@
+        ),
+        ModuleSetting(
+            key="emote_max_width",
+            label="Maximum width of emotes in pixels. Emotes exceeding this width will be counted as wide. For example: Setting this value to 130 means any emote with 131 pixels width or more will be counted as wide",
+            type="number",
+            required=True,
+            placeholder="",
+            default=128,
+            constraints={"min_value": 112, "max_value": 1000},
+        ),
+        ModuleSetting(
+            key="bypass_level",
+            label="Level to bypass module",
+            type="number",
+            required=True,
+            placeholder="",
+            default=420,
+            constraints={"min_value": 100, "max_value": 1000},
+        ),
+        ModuleSetting(
+            key="moderation_action",
+            label="Moderation action to apply",
+            type="options",
+            required=True,
+            default="Timeout",
+            options=["Delete", "Timeout"],
+        ),
+        ModuleSetting(
+            key="timeout_online",
+            label="Online timeout duration, 0 = off (if moderation action is timeout)",
+            type="number",
+            required=True,
+            placeholder="",
+            default=60,
+            constraints={"min_value": 0, "max_value": 1209600},
+        ),
+        ModuleSetting(
+            key="timeout_offline",
+            label="Offline Timeout duration, 0 = off (if moderation action is timeout)",
+            type="number",
+            required=True,
+            placeholder="",
+            default=60,
+            constraints={"min_value": 0, "max_value": 1209600},
+        ),
+        ModuleSetting(
+            key="allow_subs_to_bypass",
+            label="Allow subscribers to bypass",
+            type="boolean",
+            required=True,
+            default=False,
+        ),
+        ModuleSetting(
+            key="timeout_reason",
+            label="Timeout Reason",
+            type="text",
+            required=False,
+            placeholder="",
+            default="Too many wide emotes",
+            constraints={},
+        ),
+        ModuleSetting(
+            key="disable_warnings",
+            label="Disable warning timeouts",
+            type="boolean",
+            required=True,
+            default=False,
+        ),
+    ]
+
+    def on_message(self, source, message, emote_instances, msg_id, **rest):
+        if source.level >= self.settings["bypass_level"] or source.moderator is True:
+            return True
+
+        if self.bot.is_online and self.settings["timeout_online"] == 0:
+            return True
+
+        if not self.bot.is_online and self.settings["timeout_offline"] == 0:
+            return True
+
+        if self.settings["allow_subs_to_bypass"] and source.subscriber is True:
+            return True
+
+        wide_emotes = [1 for i in emote_instances if i.emote.max_width > self.settings["emote_max_width"]]
```

Comment:
You didn't do what I suggested - the main point was the usage of the `()` instead of `[]` - as one is an iterator, the other one constructs a new list. Maybe this stackoverflow thread would help you understand: https://stackoverflow.com/q/53376146/4464702

Reply:
I wasn't aware, let me know if I implemented it correctly. Thanks 👍

Reply:
Yeah, that looks good! Thanks!